x-pack/legacy/plugins/lens/server/routes/existing_fields.ts

```diff
@@ -82,6 +86,166 @@ export async function existingFieldsRoute(setup: CoreSetup) {
   );
 }
 
+async function fetchFieldExistence({
+  context,
+  indexPatternId,
+  fromDate,
+  toDate,
+  timeFieldName,
+}: {
+  indexPatternId: string;
+  context: RequestHandlerContext;
+  fromDate?: string;
+  toDate?: string;
+  timeFieldName?: string;
+}) {
+  const {
+    indexPattern,
+    indexPatternTitle,
+    mappings,
+    fieldDescriptors,
+  } = await fetchIndexPatternDefinition(indexPatternId, context);
+
+  const fields = buildFieldList(indexPattern, mappings, fieldDescriptors);
+
+  const docs = await fetchIndexPatternStats({
+    fromDate,
+    toDate,
+    client: context.core.elasticsearch.dataClient,
+    index: indexPatternTitle,
+    timeFieldName: timeFieldName || indexPattern.attributes.timeFieldName,
+    fields,
+  });
+
+  return {
+    indexPatternTitle,
+    existingFieldNames: existingFields(docs, fields),
+  };
+}
+
+async function fetchIndexPatternDefinition(indexPatternId: string, context: RequestHandlerContext) {
+  const savedObjectsClient = context.core.savedObjects.client;
+  const requestClient = context.core.elasticsearch.dataClient;
+  const indexPattern = await savedObjectsClient.get('index-pattern', indexPatternId);
+  const indexPatternTitle = indexPattern.attributes.title;
+  // TODO: maybe don't use IndexPatternsFetcher at all, since we're only using it
+  // to look up field values in the resulting documents. We can accomplish the same
+  // using the mappings which we're also fetching here.
+  const indexPatternsFetcher = new IndexPatternsFetcher(requestClient.callAsCurrentUser);
+  const [mappings, fieldDescriptors] = await Promise.all([
+    requestClient.callAsCurrentUser('indices.getMapping', {
+      index: indexPatternTitle,
+    }),
+
+    indexPatternsFetcher.getFieldsForWildcard({
+      pattern: indexPatternTitle,
+      // TODO: Pull this from kibana advanced settings
+      metaFields,
+    }),
+  ]);
+
+  return {
+    indexPattern,
+    indexPatternTitle,
+    mappings,
+    fieldDescriptors,
+  };
+}
```

Comment:
I'm going to see the performance hit of this, it's a bit worrying that it uses 3 network requests to get the index pattern data, and a 4th to get the existence info.